pkg/cloudprovider/cloudapi/aws/aws_converters.go

```diff
@@ -112,88 +113,92 @@ func convertFromSecurityGroupPair(cloudGroups []*ec2.UserIdGroupPair, managedSGs
 	return cloudResourceIDs, desc
 }
 
-// convertFromIPPermissionToIngressRule converts cloud ingress rules from ec2.IpPermission to internal securitygroup.IngressRule.
+// convertFromIngressIpPermissionToCloudRule converts AWS ingress rules from ec2.IpPermission to internal securitygroup.CloudRule.
 // Each AT Sg can have one or more ANPs and an ANP can have one or more rules. Each rule can have a description.
-func convertFromIPPermissionToIngressRule(ipPermissions []*ec2.IpPermission, managedSGs map[string]*ec2.SecurityGroup,
-	unmanagedSGs map[string]*ec2.SecurityGroup) []securitygroup.IngressRule {
-	var ingressRules []securitygroup.IngressRule
+func convertFromIngressIpPermissionToCloudRule(sgID string, ipPermissions []*ec2.IpPermission,
+	managedSGs, unmanagedSGs map[string]*ec2.SecurityGroup) []securitygroup.CloudRule {
+	var ingressRules []securitygroup.CloudRule
 	for _, ipPermission := range ipPermissions {
-		fromSrcIPs, desc := convertFromIPRange(ipPermission.IpRanges)
+		fromSrcIPs, descriptions := convertFromIPRange(ipPermission.IpRanges)
 		for i, srcIP := range fromSrcIPs {
 			// Get cloud rule description.
-			_, ok := securitygroup.ExtractCloudDescription(desc[i])
-			if !ok {
-				// Ignore rules that don't have a valid description field.
-				awsPluginLogger().V(4).Info("Failed to extract cloud rule description", "desc", desc[i])
-				continue
+			desc, ok := securitygroup.ExtractCloudDescription(descriptions[i])
+			ingressRule := securitygroup.CloudRule{
+				Rule: &securitygroup.IngressRule{
+					FromPort:  convertFromIPPermissionPort(ipPermission.FromPort, ipPermission.ToPort),
+					FromSrcIP: []*net.IPNet{srcIP},
+					Protocol:  convertFromIPPermissionProtocol(*ipPermission.IpProtocol),
+				},
+				AppliedToGrp: sgID,
 			}
-			var ingressRule securitygroup.IngressRule
-
-			ingressRule.FromSrcIP = []*net.IPNet{srcIP}
-			ingressRule.Protocol = convertFromIPPermissionProtocol(*ipPermission.IpProtocol)
-			ingressRule.FromPort = convertFromIPPermissionPort(ipPermission.FromPort, ipPermission.ToPort)
-
+			if ok {
```

Comment:
is this required to continue having rules without description and let sync with cloud delete the rule??

Reply:
It's not for sync to delete the rule since these rules will be preserved as user rules. This is for sync to update cloudRule indexer for rule duplicate computation.

Reply:
collision with user created rule? could you check if azure allow duplicate rule with different rule priority ? for example allow ssh from X with priority X and allow ssh from X with priority Y.

Reply:
On Azure it actually does. Just needs to be different priority AND different rule name.